src/webapi/events/mouse.rs

```diff
@@ -439,6 +439,87 @@ impl ConcreteEvent for MouseLeaveEvent {
+/// is rotated over the element that has the listener attached.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/Events/wheel)
+// https://w3c.github.io/uievents/#event-type-wheel
+#[derive(Clone, Debug, PartialEq, Eq, ReferenceType)]
+#[reference(instance_of = "MouseEvent")] // TODO: Better type check.
+#[reference(subclass_of(Event, UiEvent, MouseEvent))]
+pub struct MouseWheelEvent( Reference );
+
+impl IEvent for MouseWheelEvent {}
+impl IUiEvent for MouseWheelEvent {}
+impl IMouseEvent for MouseWheelEvent {}
+impl ConcreteEvent for MouseWheelEvent {
+    const EVENT_TYPE: &'static str = "wheel";
+}
+
+impl MouseWheelEvent {
+    /// The change in X of the wheel
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent/deltaX)
+    // https://w3c.github.io/uievents/#dom-wheelevent-deltax
+    pub fn delta_x(&mut self) -> f64 {
+        js! (
+            @{&self.0}.deltaX
+        ).try_into().unwrap()
+    }
+
+    /// The change in Y of the wheel
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent/deltaY)
+    // https://w3c.github.io/uievents/#dom-wheelevent-deltay
+    pub fn delta_y(&mut self) -> f64 {
+        js! (
+            @{&self.0}.deltaY
+        ).try_into().unwrap()
+    }
+
+    /// The change in Z of the wheel
+    ///
+    /// [(Javascript docs)](https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent/deltaZ)
+    // https://w3c.github.io/uievents/#dom-wheelevent-deltaz
+    pub fn delta_z(&mut self) -> f64 {
+        js! (
+            @{&self.0}.deltaZ
+        ).try_into().unwrap()
+    }
+
+    /// The unit of measure of change
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent/deltaMode)
+    // https://w3c.github.io/uievents/#dom-wheelevent-deltamode
+    pub fn delta_mode(&mut self) -> MouseWheelDeltaMode {
+        let mode: u64 = js! (
+            @{&self.0}.deltaMode
+        ).try_into().unwrap();
+        match mode {
+            0 => MouseWheelDeltaMode::DomDeltaPixel,
+            1 => MouseWheelDeltaMode::DomDeltaLine,
+            2 => MouseWheelDeltaMode::DomDeltaPage,
+            _ => unreachable!()
+        }
+    }
+}
+
+/// What unit of measure the mouse wheel delta is in
+///
+/// [(JavaScipt docs)](https://developer.mozilla.org/en-US/docs/Web/API/WheelEvent/deltaMode)
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum MouseWheelDeltaMode {
+    /// The unit of measurement for the delta is pixels
+    // https://w3c.github.io/uievents/#dom-wheelevent-dom_delta_pixel
+    DomDeltaPixel = 0,
+     /// The unit of measurement for the delta is lines
+    // https://w3c.github.io/uievents/#dom-wheelevent-dom_delta_line
+    DomDeltaLine = 1,
+     /// The unit of measurement for the delta is pages
+    // https://w3c.github.io/uievents/#dom-wheelevent-dom_delta_page
+    DomDeltaPage = 2
```

Comment:
1. Get rid of `= <number>` to make the enum opaque.
2. We can probably delete the `DomDelta` prefix.